### PR TITLE
#42, 40, 38, 39, 80 User Interface

### PR DESCRIPTION
**Pull request by smeas**
Task 80, 38, 39, 40, 24

hNgY7ueWe8

### CODE REVIEW

**Review by Addehman — APPROVED** [review]
Noice!

File: Assets/PauseMenu.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿using UnityEngine;
+using UnityEngine.Events;
+
+public class PauseMenu : MonoBehaviour {
```
Comment (MrJackey) [inline Assets/PauseMenu.cs:4]: This should probably be moved into a specific folder.

Comment (MrJackey) [inline Assets/PauseMenu.cs]: This script should probably be moved into a specific folder.

File: Assets/Scripts/UI/TitleScreen.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using UnityEngine;
+
+public class TitleScreen : MonoBehaviour {
+	[SerializeField] private MenuScreens mainMenu;
+	[SerializeField] private GameObject anyKeyPrompt;
+
+	private void Update() {
+		if (Input.anyKey) {
```
Comment (MrJackey) [inline Assets/Scripts/UI/TitleScreen.cs]: Does this include clicking on a gamepad?

File: Assets/Scripts/UI/UIHealthSetter.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		if (go != null)
+			playerHealth = go.GetComponent<Health>();
+
+		if (playerHealth != null) {
+			bar.MaxValue = playerHealth.MaxValue;
+			bar.Value = playerHealth.Value;
+		}
+	}
+
+	private void Update() {
```
Comment (MrJackey) [inline Assets/Scripts/UI/UIHealthSetter.cs]: Would it be fitting to use lateupdate here to really get the player's hp?

**Review by MrJackey — CHANGES_REQUESTED** [review]
Just a few questions, looks good in general though

File: Assets/PauseMenu.cs (hunk 1 above)
Comment (smeas) [reply Assets/PauseMenu.cs]: Whoops.

File: Assets/Scripts/UI/TitleScreen.cs (hunk 2 above)
Comment (smeas) [reply Assets/Scripts/UI/TitleScreen.cs]: I looked it up and indeed, this only works for keyboard and mouse. There seems to be no equivalent for gamepads in the built-in input system. I'm not quite sure how to go about making this work with gamepads yet. Any ideas?

File: Assets/Scripts/UI/UIHealthSetter.cs (hunk 3 above)
Comment (smeas) [reply Assets/Scripts/UI/UIHealthSetter.cs]: Good idea!

**Review by MrJackey — APPROVED** [review]
Looks good

**Review by MrJackey — APPROVED** [review]

**Pull request merged**